### PR TITLE
feat: Add some backend tests

### PR DESCRIPTION
**Pull request by MateuszKikmunter**
I added unit tests for the backend.
Now we have all controllers, most of the helpers and all middleware tested. 
Additionally I also removed some unsed code and files and made some refactoring to make the code more modern.
Next step would be to add some integration tests to be fully covered. (I can work on that too if you agree @gitfrosh )

### CODE REVIEW

File: backend/controllers/api.js [hunk 1]
```diff
@@ -1,66 +0,0 @@
-const Book = require("./../models/book.model");
```
Comment (EddieEldridge) [inline backend/controllers/api.js:1]: How is it that we can just remove this file? Is it not being used?

**Comment by EddieEldridge** [comment]
Had a quick look and all goods to me. This will be very helpful. If I had any feedback it would be

- Move tests into their own dedicated folder (personal preference)
- Add some documentation on how to run the tests to the README
- Update package.json with the correct command to run the tests

**Review by EddieEldridge — APPROVED** [review]

File: backend/controllers/api.js (hunk 1 above)
Comment (MateuszKikmunter) [reply backend/controllers/api.js:1]: Nope, not used at all. Only specialized controllers like book, quote etc. are used. You can checked exposed endpoint in the [docs](https://the-one-api.dev/documentation#4).

**Comment by MateuszKikmunter** [comment]
> Had a quick look and all goods to me. This will be very helpful. If I had any feedback it would be
> 
> * Move tests into their own dedicated folder (personal preference)
> * Add some documentation on how to run the tests to the README
> * Update package.json with the correct command to run the tests (if the existing one doesnt do it already)

Thanks for input @EddieEldridge !
- I followed the genral convention and placed the spec files close to the actual implementations. 
- There's a command for running tets in the package.json file, just go to the backend and run `npm test`.
- Good point about `README`, thanks! :)

**Comment by EddieEldridge** [comment]
Any chance this could get merged? It looks good to me and there's a few other bits I would like to work on using this functionality. Thanks.

**Comment by MateuszKikmunter** [comment]
> Any chance this could get merged? It looks good to me and there's a few other bits I would like to work on using this functionality. Thanks.

Same here. There are few thing I’d like to take a look at after this PR is merged.

**Comment by gitfrosh** [comment]
Thank you @MateuszKikmunter and @EddieEldridge, for your contributions and reminders regarding the pull request! Finally, a proper test setup for the API 👍🏻  I'll merge this now, and you're very welcome to proceed with your work.

**Pull request merged**